### PR TITLE
fix krlmlr/r-appveyor#42

### PR DESCRIPTION
**Pull request by wush978**

### CODE REVIEW

**Comment by wush978** [comment]
The PR for #42

**Comment by wush978** [comment]
I think it works now. The test log does use the gcc-4.6.3 as shown in <https://ci.appveyor.com/project/krlmlr/r-appveyor/build/1.0.805#L135>

File: scripts/appveyor-tool.ps1 [hunk 1]
```diff
@@ -93,7 +93,23 @@ Function Bootstrap {
   echo "Rtools is now available on drive $RtoolsDrive"
 
   Progress "Setting PATH"
-  $env:PATH = $RtoolsDrive + '\Rtools\bin;' + $RtoolsDrive + '\Rtools\MinGW\bin;' + $RtoolsDrive + '\Rtools\gcc-4.6.3\bin;' + $env:PATH
+  if ( -not(Test-Path Env:\GCC_PATH) ) {
+    echo "Detecting whether C++11 is required"
+    $ErrorActionPreference = "Continue"
+    Rscript -e "stopifnot(grepl('[cC]\\+\\+11', read.dcf('DESCRIPTION')[,'SystemRequirements']))"
+    if ($LastExitCode -ne 0) {
+      $gcc_path = "gcc-4.6.3"
```
Comment (krlmlr) [inline scripts/appveyor-tool.ps1]: Sorry it took me so long.

I really think we should keep things simple and stupid here, and use this alternative if `GCC_PATH` is not specified; at some later point, the default might become "mingw_32".

Comment (wush978) [inline scripts/appveyor-tool.ps1]: OK. I'll modify the PR this weekend.

**Comment by krlmlr** [comment]
Thanks!

**Comment by krlmlr** [comment]
@wush978: For some reason, the "fakepackage" used for testing fails when building with GCC_PATH=mingw_32: https://ci.appveyor.com/project/krlmlr/r-appveyor/build/1.0.826#L244. Any idea what might be wrong here?

**Comment by wush978** [comment]
@krlmlr: I'll check it out this weekend.

**Comment by wush978** [comment]
@krlmlr 

Because the new tool chain gcc-4.9.3 on windows does not support multilib, so we need to use different compiler for 32bit and 64bit. I am still wondering how to set PATH or other environment variables properly to make R use two compilers.

Reference: <https://github.com/rwinlib/r-base#migration-steps-for-cran>

**Comment by wush978** [comment]
Maybe we should solve this issue via:

1. Remove `c:\Rtools\gcc-4.6.3\bin` from the environment variable PATH.
1. Edit the compiler PATH to `C:\Program Files\R\R-3.2.2\etc\i386\Makeconf` and `C:\Program Files\R\R-3.2.2\etc\x64\Makeconf` properly.
    - The variable `BINPREF` should be set to the directory of the compiler

For example, for gcc-4.6, the content of both `C:\Program Files\R\R-3.2.2\etc\i386\Makeconf` and `C:\Program Files\R\R-3.2.2\etc\x64\Makeconf` should be:

```
## Things which are substituted by fixed/Makefile (and also -O3 -> -O2)
WIN = 32
BINPREF = c:/Rtools/gcc-4.6.3/bin/
# SYMPAT = 's/^.* [BCDRT] _/ /p' for 32-bit
```

For gcc-4.9.3, the content of `C:\Program Files\R\R-3.2.2\etc\i386\Makeconf` should be:

```
## Things which are substituted by fixed/Makefile (and also -O3 -> -O2)
WIN = 32
BINPREF = c:/Rtools/mingw_32/bin/
# SYMPAT = 's/^.* [BCDRT] _/ /p' for 32-bit
```

And the content of `C:\Program Files\R\R-3.2.2\etc\x64\Makeconf` should be:

```
## Things which are substituted by fixed/Makefile (and also -O2 -> -O2)
WIN = 64
BINPREF = c:/Rtools/mingw_64/bin/
# SYMPAT = 's/^.* [BCDRT] _/ /p' for 32-bit
```

I do not know if this is the recommended way to solve the issue. How about asking them on [r-sig-windows](https://stat.ethz.ch/mailman/listinfo/r-sig-windows)?

**Comment by krlmlr** [comment]
Current master uses --no-multiarch and enables other tweaks, see README.

**Pull request merged**